tests/test_updater_ng.py

```diff
@@ -326,6 +333,109 @@ def test_non_existing_target_file(self) -> None:
         with self.assertRaises(exceptions.DownloadHTTPError):
             self.updater.download_target(info)
 
+    # TAP 14 tests ~ REMOVE COMMENT BEFORE THE NEXT COMMIT!!
+
+    # test case to check for the TAP 14 folder
+    def test_check_folder_tap14(self) -> None:
+        # Creating the parent folder for the TAP 14 folder
+        self.assertTrue(os.path.isdir(self.tap14_directory))
+
+    def test_check_tap14_contents(self) -> None:
+        # Checking specific files inside TAP 14
+        filenames = ["targets.json", "root.json"]
+        for file in filenames:
+            self.assertTrue(
+                os.path.isfile(os.path.join(self.tap14_directory, file))
+            )
+
+        # Checking specific folders inside TAP 14
+        foldernames = ["targets", "1.0.0"]
+        for folder in foldernames:
+            self.assertTrue(
+                os.path.isdir(os.path.join(self.tap14_directory, folder))
+            )
+
+    def test_get_spec_version_supported(self) -> None:
+        """This uses the default SUPPORTED_VERSIONS variable from updater.py"""
+
+        with self.assertRaises(
+            exceptions.RepositoryError,
+            msg="Latest repository version less than 4",
+        ):
+            _get_spec_version(
+                ["1", "2", "3"], "4", ngclient.updater.SUPPORTED_VERSIONS
+            )
+
+        self.assertEqual(
+            _get_spec_version(
+                ["1", "2", "3"], "3", ngclient.updater.SUPPORTED_VERSIONS
+            ),
+            ("3", None),
+            "3 is selected as the spec version and no warning ensues",
+        )
+
+    def test_get_spec_version(self) -> None:
```

Comment:
I think now you need to account for the `None` case for `repository_versions`

This is actually starting to get a little confusing: there's "version/path pairs" (which come from the repository versions, and then there are just "versions" (from supported versions). `_get_spec_version` should pick a matching (version, path) pair.

It might be worth encoding that in the type system.

Reply:
Or maybe even just have a function from "path" to version and use that in `_get_spec_version`.